debezium-api/src/main/java/io/debezium/spi/snapshot/SnapshotterProvider.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.spi.snapshot;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This interface is used to provide custom snapshotters:
+ * Implementations must:
+ *
+ * provide a map of snapshotter in the {@link #create(Configuration config)} method.
+ *
+ * @author Mario Fiore Vitale
+ */
+public interface SnapshotterProvider {
+
+    /**
+     * Create a map of snapshotter where the key is its name used in 'snapshot.mode' configuration.
+     *
+     * @param Configuration the connector configuration
+     *
+     * @return a map of custom snapshotter
+     */
+    Map<String, Snapshotter> create(Configuration config); // Can we move the Configuration interface from core module?
```

Comment:
I'd be very careful with moving anything which includes Kafka stuff into the api package. Especially when it's even not a part of Kafka API package (e.g. `connect-api`). I'd prefer to keep it int core package for now.

Reply:
This `Configuration` is `io.debezium.config.Configuration` so is not Kafka related. 

In any case I'm evaluating the Chris approach and maybe in that case we don't need it.

Reply:
> This `Configuration` is `io.debezium.config.Configuration` so is not Kafka related.

Yes, but if you look into the `io.debezium.config.Configuration` class [here](io.debezium.config.Configuration), the class itself contains dependency on Kafka `org.apache.kafka.common.config.ConfigValue`, so by moving `io.debezium.config.Configuration` into api module, the module will start to depend on Kafka, which I'm not sure we want it (at least personally I don't like it). 

> In any case I'm evaluating the Chris approach and maybe in that case we don't need it.

yes, this would be IMHO nice example where this dependency mechanism would be very useful - we can avoid loading user configuration here and thus avoid various kind of dependencies (which is always good, no matter if I like it or not:-)